### PR TITLE
CLI Output Pre-Processors

### PR DESCRIPTION
**Pull request by dongilbert**
Following the discussion on #279, I felt it necessary to add support for other types of output pre-processors into the system. This PR does that.

### CODE REVIEW

**Comment by asika32764** [comment]
:+1:

**Comment by piotr-cz** [comment]
is the value of @since docblock `__DEPLOY_VERSION__` intentional?

**Comment by elkuku** [comment]
> is the value of @since docblock **DEPLOY_VERSION** intentional?

That seems to be a common problem for PRs towards the different Joomla! projects :wink: 
Maybe we can do some script that replaces those tags **on merge**....

Looks sweet :heart:

File: src/Joomla/Application/Cli/Output/Processors/ColorProcessor.php [hunk 1]
```diff
@@ -0,0 +1,180 @@
+<?php
+/**
+ * Part of the Joomla Framework Application Package
+ *
+ * @copyright  Copyright (C) 2005 - 2013 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Application\Cli\Output\Processors;
```
Comment (florianv) [inline src/Joomla/Application/Cli/Output/Processors/ColorProcessor.php]: Are we using pluralized namespaces ? IMO we should set a convention for this.

Some examples (singularized) :

https://github.com/joomla/joomla-framework/tree/staging/src/Joomla/Profiler/Renderer
https://github.com/joomla/joomla-framework/tree/staging/src/Joomla/Registry/Format

**Pull request merged**